mongo/integration/unified/server_api_options_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package unified
+
+import (
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// ServerAPIOptions is a wrapper for *options.ServerAPIOptions. This type implements the bson.Unmarshaler interface
+// to convert BSON documents to a ServerAPIOptions instance.
+type ServerAPIOptions struct {
+	*options.ServerAPIOptions
+}
+
+type ServerAPIVersion = options.ServerAPIVersion
+
+var _ bson.Unmarshaler = (*ServerAPIOptions)(nil)
+
+func (to *ServerAPIOptions) UnmarshalBSON(data []byte) error {
```

Comment:
nit: I assume the receiver is named `to` because this is copied from `TransactionOptions`. Can we rename to `s` so it more closely matches the name of the type?

Reply:
Caught red-handed! Renamed to `s`.